php/flights.php

```diff
@@ -141,20 +149,41 @@
 if (!$sth->execute($params)) {
     die('Error;Query ' . print_r($_GET, true) . ' caused database error ' . $sql . ', ' . $sth->errorInfo()[0]);
 }
-$first = true;
 
 if ($export == "export" || $export == "backup") {
     // Start with byte-order mark to try to clue Excel into realizing that this is UTF-8
     print "\xEF\xBB\xBFDate,From,To,Flight_Number,Airline,Distance,Duration,Seat,Seat_Type,Class,Reason,Plane,Registration,Trip,Note,From_OID,To_OID,Airline_OID,Plane_OID\r\n";
 }
-$gcmap_city_pairs = ''; // list of city pairs when doing gcmap export.
+
+$rows = [];
 foreach ($sth as $row) {
+    $src_apid = $row["src_apid"];
+    $src_code = format_apcode2($row["src_iata"], $row["src_icao"]);
+
+    $dst_apid = $row["dst_apid"];
+    $dst_code = format_apcode2($row["dst_iata"], $row["dst_icao"]);
+
+    if ($row["opp"] == 'Y') {
+        $tmp = $src_apid;
+        $src_apid = $dst_apid;
+        $dst_apid = $tmp;
+
+        $tmp = $src_code;
+        $src_code = $dst_code;
+        $dst_code = $tmp;
+    }
+
+    if ($export == "gcmap") {
```

Comment:
gcmap only used `$dst_code` and `$src_code` so might aswell do this earlier. The inversion is about the only thing that changes the gcmap output...

It's not going to be a route, so save a few operations by getting this done and out of the way.